### DISCUSSION TITLE
How to avoid the redundancy of running TypeScript binaries under `@yarnpkg/pnpify`?

### DISCUSSION THREAD

**Discussion by wizardlink** [yarnpkg/berry#5069] [Q&A]
I'll be as brief as I can, considering I am quite mad at this lack of documentation and explanation as to why this is needed and why it even is an issue.

On a clean project I've added: `typescript`, `@types/node` and `@yarnpkg/pnpify`, my entry point is `out/index.js` and the source file is `src/index.ts`.

In the source file all I have added is a simple `console.log(':)')` and to transpile it I have ran `yarn tsc -p ./` which that spews me the error:
```console
src/index.ts:1:1 - error TS2584: Cannot find name 'console'. Do you need to change your target library? Try changing the 'lib' compiler option to include 'dom'.

1 console.log("haha")
  ~~~~~~~


Found 1 error in src/index.ts:1
```

If I run under `pnpify` (`yarn pnpify tsc -p ./`) I get these obscure nonsense:
```console
.yarn/cache/@types-emscripten-npm-1.39.6-c9c4021365-437f2f9cdf.zip/node_modules/@types/emscripten/index.d.ts:53:33 - error TS2304: Cannot find name 'WebGLRenderingContext'.

53     preinitializedWebGLContext: WebGLRenderingContext;
                                   ~~~~~~~~~~~~~~~~~~~~~

.yarn/cache/@types-emscripten-npm-1.39.6-c9c4021365-437f2f9cdf.zip/node_modules/@types/emscripten/index.d.ts:67:28 - error TS2304: Cannot find name 'MessageEvent'.

67     onCustomMessage(event: MessageEvent): void;
                              ~~~~~~~~~~~~


Found 2 errors in the same file, starting at: .yarn/cache/@types-emscripten-npm-1.39.6-c9c4021365-437f2f9cdf.zip/node_modules/@types/emscripten/index.d.ts:53
```

Then I decide, ok, I'll explicitly add in my `tsconfig.json` that `node` is one of the typings I have, but when I run `yarn tsc` or even directly `.yarn/sdks/typescript/bin/tsc` I get the following:
```console
error TS2688: Cannot find type definition file for 'node'.
  The file is in the program because:
    Entry point of type library 'node' specified in compilerOptions

  tsconfig.json:15:15
    15     "types": ["node"]
                     ~~~~~~
    File is entry point of type library specified here.


Found 1 error.
```

But under `pnpify` in this case it transpiles with no complaints whatsoever, so my question is: Why is the sdk that yarn bundles has this issue? From what I can tell too when talking to my friends, no one that is running under Windows has this issue but some people in Linux do, but not everyone.

Does anyone knows why this is the case and if there is a workaround? Because having to explain to research and explain how to integrate `tsserver`, `eslint`, `prettier` and any other toolset in each kind of editor we may use through this will be hell.

**Comment by RichPersonM** [comment]
Have you found a solution?

**Reply by wizardlink** [reply]
We dropped berry shortly after, might not be a problem nowadays.

**Comment by francis-Paul-code** [comment]
This is still a problem, i dont think theres any attempt by the devs to fix it

**Reply by clemyan** [reply]
`@yarnpkp/pnpify` has been mostly discontinued in favor of `@yarnpkg/sdks`. Are you encountering issues when using the SDK? If so, please submit a bug report with reproduction and we can take a look.
